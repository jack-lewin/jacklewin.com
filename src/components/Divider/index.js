import React from 'react';

import styles from './styles.module.scss';

const Divider = () => <hr className={styles.divider} />;

export default Divider;
