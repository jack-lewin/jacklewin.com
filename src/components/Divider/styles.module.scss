@import "../../functions.scss";

.divider {
  width: 60%;
  width: calc(100% - (400px + #{spacing(x-large)}));
  margin: spacing(x-large) 0;
  border-width: 0;
  border-bottom-width: 1px;
}
