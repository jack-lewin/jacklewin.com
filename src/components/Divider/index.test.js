import React from 'react';
import { shallow } from 'enzyme';
import Divider from '.';

describe('Divider', () => {
  it('should render as expected', () => {
    const element = shallow(<Divider />);
    expect(element).toMatchSnapshot();
  });
});
