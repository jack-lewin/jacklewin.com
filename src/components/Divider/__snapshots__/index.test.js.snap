// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Divider />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "divider",
    },
    "ref": null,
    "rendered": null,
    "type": "hr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "divider",
      },
      "ref": null,
      "rendered": null,
      "type": "hr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
